redaxo/src/addons/phpmailer/lib/mailer.php

```diff
@@ -56,19 +56,44 @@ public function send()
             if ($this->log) {
                 $this->log();
             }
-            return parent::send();
+            $addon = rex_addon::get('phpmailer');
+            if(!parent::send() && $addon->getConfig('logging') != 0){
+            $this->toMailerLog('ERROR');
+            return false;
+            }
+       
+        if($addon->getConfig('logging') == 2)
+        {
+        $this->toMailerLog('OK');
+        }
+        return true;
         });
     }
+    /**
+     * @param string $success
+    */
+    public function toMailerLog($succsess)
+    {
+        $log = new rex_log_file(rex_path::addonData('phpmailer', 'mail.log'), 2000000);
+        $data = [
+            ($succsess),
+            ($this->From.PHP_EOL),
+            (implode(', ', array_column($this->getToAddresses(), 0)).PHP_EOL),
+            (str_replace('|',':pipe:',$this->Subject.PHP_EOL)),
+            strip_tags($this->ErrorInfo),
+        ];
+        $log->add($data);
+    }    
 
     /**
      * @param bool $status
      */
     public function setLog($status)
     {
-        $this->log = $status;
+        $this->mail_archive = $status;
     }
 
-    private function log()
+    private function mail_archive()
```

Comment:
Methodennamen immer in camelCase.

Reply:
lautet jetzt nur noch archive(), danke für den Hinweis